src/generic/datavgen.cpp

```diff
@@ -3192,6 +3356,303 @@ int wxDataViewMainWindow::GetLineHeight( unsigned int row ) const
 }
 
 
+void RowRange::Add(const unsigned int idx)
+{
+    RangeEntity *rng = NULL;
+
+    size_t count = m_ranges.GetCount();
+    size_t rngIdx = 0;
+    while (rngIdx < count)
```

Comment:
This loop should be written as

    for ( unsigned rngIdx = 0; rngIdx < count; ++rngIdx )

as this is exactly what it does and seeing `while` here is confusing as you expect something more complicated than this, precisely because the loop is not written using just a simple `for`.